adapter/src/dbt/adapters/fal_experimental/utils.py

```diff
@@ -80,3 +84,54 @@ def db_adapter_config(config: RuntimeConfig) -> RuntimeConfig:
         new_config = config
 
     return new_config
+
+
+def load_environments(base_dir: str) -> Dict[str, "BaseEnvironment"]:
+    import os
+    fal_project_path = os.path.join(base_dir, "fal_project.yml")
+    if not os.path.exists(fal_project_path):
+        raise FalParseError(f"{fal_project_path} must exist to define environments")
+
+    fal_project = load_yaml(fal_project_path)
+    environments = {}
+    for environment in fal_project.get("environments", []):
+        env_name = _get_required_key(environment, "name")
+        if _is_local_environment(env_name):
+            raise FalParseError(
+                f"Environment name conflicts with a reserved name: {env_name}."
+            )
+
+        env_kind = _get_required_key(environment, "type")
+        environments[env_name] = create_environment(env_name, env_kind, environment)
+    return environments
```

Comment:
We introduced this yesterday: https://github.com/fal-ai/fal/pull/605/files#diff-123d19b616a2d38275554b5d9bfdea71bd79d5eabc31e98ca8d6233c9f423e60R234

In dbt-fal's case, it probably makes sense to add it to `fetch_environment` where we use `load_environments` and replace 
```
RuntimeException(str(exc)) from exc
```
with
```
RuntimeException("Error loading environments from fal_project.yml") from exc
```